src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -138,14 +143,15 @@ bool CLeastAngleRegression::train_machine(CFeatures* data)
 		for (int32_t j=0; j < n_fea; ++j)
 			X(i,j) = Xorig(j,i);
 	}
+	Map<MatrixXd> map_X(X.matrix, n_vec, n_fea);
 
 	// beta is the estimator
 	vector<float64_t> beta = make_vector(n_fea, 0);
 
 	vector<float64_t> Xy = make_vector(n_fea, 0);
+	Map<VectorXd> map_Xy(&Xy[0], n_fea);
 	// Xy = X' * y
-	cblas_dgemv(CblasColMajor, CblasTrans, n_vec, n_fea, 1, X.matrix, n_vec,
-		y.vector, 1, 0, &Xy[0], 1);
+	map_Xy=map_X.transpose()*map_y;
```

Comment:
so here basically we X^T^T, i.e. double transpose the original input..... so we could just use the input itself right? :)